### ISSUE TITLE
Refactor: add support for standardize transform directly within Emulator

### ISSUE THREAD

**Issue by sgreenbury** [alan-turing-institute/autoemulate#645]
From discussion with @radka-j and relating to #618, an option to increase the intuitiveness for working with emulators would be to include `StandardizeTransform` within the emulators (optionally) instead of requiring `TransformedEmulator`.

This could be implemented directly in the `Emulator` within fit and predict.

The transforms included could be generalized to anything that does not break the gradients being able to be taken.

We could also extend `TransformedEmulator` to be a `nn.Module` (even with a model that is not `nn.Module`) to ensure gradients can be taken when also working with dim reduction but this could be opened as a separate non-refactor issue.

**Comment by sgreenbury** [comment]
Also remove the default setting regarding max targets in transform so that sampling isn't triggered automatically.

**Comment by sgreenbury** [comment]
From discussion with @radka-j:
- Aim to support standardize across all emulators (since it is expected to only improve fitting and does not stop other transforms being used first)
- We'll aim to default to standardize x and y for GPs by default as true as fitting will not work in most cases without this. To then use within the AE compare loop the GP could be subclassed
```python
class GPWithoutStandardization(GaussianProcessExact):
    def __init__(self, x, y, standardize_x=False, standardize_y=False, **kwargs):
        super().__init__(self, x, y, standardize_x, standardize_y, **kwargs):
```
- For the other emulators for now we'll default to `False` in the init
